### DISCUSSION TITLE
Passing the close function from Dialog to the Modal component

### DISCUSSION THREAD

**Discussion by ziyak97** [adobe/react-spectrum#4707] [Q&A] [answered]
I’m using the SlideUpModal component which requires an onClose prop. However, I can only access the close function inside the Dialog component which needs to be nested under the Modal. How can I pass the close function from the Dialog to the onClose prop of the SlideUpModal component?

```
<DialogTrigger>
  <Button>Open Modal</Button>
  <SlideUpModal isDismissable>
    <Dialog>
      {({ close }) => (
        <>
          <p>Content of the modal</p>
          <Button onPress={close}>Close</Button>
        </>
      )}
    </Dialog>
  </SlideUpModal>
</DialogTrigger>
```

In this example, we need to pass the close function from the Dialog to the onClose prop of the SlideUpModal, but it’s not clear how to do that.

**Comment by reidbarber** [comment]
I'm not sure what the SlideUpModal component is, so it may be useful to post a codesandbox to give some additional context. Maybe you can consider making it [controlled](https://react-spectrum.adobe.com/react-aria/Modal.html#controlled-open-state) instead?

**Reply by ziyak97** [reply]
if i make it controlled I'll no longer need dialog trigger and dialog and will be able to open and close it programmatically? on a side not since native dialog is now widely supported would there be an eventual transition to that?

**Reply by reidbarber** [reply]
Correct, you can just set the controlled state with the Button's onPress. And I don't think we will switch to a native dialog soon since it would be a breaking change, and it may prevent normalizing some features across browsers (focus, styling, scrolling, etc.).

**Reply by ziyak97** [reply]
okay thanks a ton I'll give this a shot tomorrow and report back on if it works 😁

**Reply by snowystinger** [reply, marked as answer]
Couple other things to try, you could look at our nightlies for this functionality https://github.com/adobe/react-spectrum/pull/4684
You could also try inverting the order and wrap the `SlideUpModal` inside the `Dialog`

**Reply by ziyak97** [reply]
just inverting it worked 😁. i actually tried this earlier but was getting unexpected results which turned out to be me just messing up my SlideUpModal code. thanks a ton!

**Comment by ziyak97** [comment]
my issue had been resolved check the comments on the post for further details.
